rest/server_context.go

```diff
@@ -254,9 +280,11 @@ func (sc *ServerContext) Close(ctx context.Context) {
 	sc.invalidDatabaseConfigTracking.dbNames = nil
 
 	for _, s := range sc._httpServers {
-		base.InfofCtx(ctx, base.KeyHTTP, "Closing HTTP Server: %v", s.addr)
-		if err := s.server.Close(); err != nil {
-			base.WarnfCtx(ctx, "Error closing HTTP server %q: %v", s.addr, err)
+		if s.server != nil {
```

Comment:
I'm not opposed to doing a nil check here, but I don't see how this would end up nil, and I'm worried we don't do the nil checks in the other places we need.

I assume that this is from the rest tester usage.

Reply:
Yeah, `RestTester` doesn't start up an actual HTTP server, so falling into this code now results in a panic because we populate a "fake" list of servers on ServerContext for the reason of grabbing `serverInfo` for stamping `s.addr`